core/math/random_pcg.cpp

```diff
@@ -42,6 +43,26 @@ void RandomPCG::randomize() {
 	seed(((uint64_t)OS::get_singleton()->get_unix_time() + OS::get_singleton()->get_ticks_usec()) * pcg.state + PCG_DEFAULT_INC_64);
 }
 
+int RandomPCG::rand_weighted(const Vector<float> &p_weights) {
+	ERR_FAIL_COND_V_MSG(p_weights.is_empty(), 0, "Weights array is empty.");
+	int32_t weights_size = p_weights.size();
+	const float *weights = p_weights.ptr();
+	float weights_sum = 0.0;
+	for (int i = 0; i < weights_size; ++i) {
+		weights_sum += weights[i];
+	}
+
+	float remaining_distance = Math::randf() * weights_sum;
+	for (int i = 0; i < weights_size; ++i) {
```

Comment:
```suggestion
	int64_t weights_size = p_weights.size();
	const float *weights = p_weights.ptr();
	float weights_sum = 0.0;
	for (int64_t i = 0; i < weights_size; ++i) {
		weights_sum += weights[i];
	}

	float remaining_distance = Math::randf() * weights_sum;
	for (int64_t i = 0; i < weights_size; ++i) {
```
My bad forgot the type here

Reply:
no worries. good catch. done!